src/index_emails.py

```diff
@@ -82,24 +84,74 @@ def create_index():
 def upload_batch(upload_data):
     upload_data_txt = ""
     for item in upload_data:
-        cmd = {'index': {'_index': tornado.options.options.index_name, '_type': 'email', '_id': item['message-id']}}
+        cmd = {'index': {'_index': tornado.options.options.index_name, '_type': tornado.options.options.es_document_type, '_id': item['message-id']}}
         upload_data_txt += json.dumps(cmd) + "\n"
         upload_data_txt += json.dumps(item) + "\n"
 
-    request = HTTPRequest(tornado.options.options.es_url + "/_bulk", method="POST", body=upload_data_txt, request_timeout=240)
-    response = http_client.fetch(request)
-    result = json.loads(response.body)
+    upload_successful = False
+    upload_attempts = 0
+    res_txt = ""
+    while ((not upload_successful) and (upload_attempts < tornado.options.options.es_upload_attempts)):
```

Comment:
Why is this retry logic needed?

Reply:
generally I prefer something like
```
attempts_left = 10
while attempts_left:
    # do something
    if success:
         break
    attempts_left -= 1
```

Reply:
Permits flexibility for transient network failures or load in ES where the request fails to complete within timeout.  Such as `Cannot assign requested address` failures, something with the http client lib you are using is producing for very large mbox files that take a while to run. Looking more into it.

Reply:
Ok

Reply:
If you don't intend to fix/improve this section in this PR, can you please revert the changes in `upload_batch()`?